src/posix/client.cpp

```diff
@@ -76,95 +76,132 @@ static void InputCallback(char *aLine)
 }
 #endif // OPENTHREAD_USE_READLINE
 
-static bool FindPrompt(int &aState, char aChar)
+enum PromptState
+{
+    kPromptStateNewline,
+    kPromptStateChar1Received,
+    kPromptStatePrompted,
+    kPromptStateWaitNewline,
+};
+
+static bool FindPrompt(PromptState &aState, char aChar)
 {
     switch (aChar)
     {
     case '>':
-        aState = aState == 0 ? 1 : -1;
+        aState = aState == kPromptStateNewline ? kPromptStateChar1Received : kPromptStateWaitNewline;
```

Comment:
For state machines, I would prefer if we "pivot" this switch statement to `switch (aState)`. It makes it easier to ensure we handle each state appropriately and the conditions for state transmissions from each state.

Reply:
Agree.